### PR TITLE
add codeless execution

### PR DESCRIPTION
**Pull request by itayka**
Please Make sure these boxes are checked before submitting your pull request - Thanks ahead!

- [ ] Proper pull request title - concise and clear for others.
- [ ] Proper pull request short description - clear and concise (as it should appear in the Jira ticket) for other developers and users.
- [ ] Proper Jira ticket - Number, Link in pull request description.
- [ ] The PR can is merged on your machine without any conflicts.
- [ ] The PR can is built on your machine without any (new) warnings.
- [ ] The PR passed sanity tests by you / QA / DevTest / Good Samaritan.
- [ ] Add unit tests with new features.
- [ ] If you added any dependency to the POM - Please update grount

### CODE REVIEW

File: src/main/java/com/microfocus/application/automation/tools/octane/executor/TestExecutionJobCreatorService.java [hunk 1]
```diff
@@ -351,12 +353,20 @@ public static FreeStyleProject createExecutor(DiscoveryInfo discoveryInfo) {
...
+			if (isMbt) { // in case of MBT add a second runner for codeless
+				codelessRunner = new RunFromCodelessBuilder();
+			}
+			// add steps to project
+			proj.getBuildersList().add(converterBuilder);
 			proj.getBuildersList().add(uftRunner);
-
+			if(codelessRunner != null) {
+				proj.getBuildersList().add(codelessRunner);
+			}
```
Comment (nissimshitrit) [inline src/main/java/com/microfocus/application/automation/tools/octane/executor/TestExecutionJobCreatorService.java:365]: please re-organize code, all added code can be done under "isMbt"

File: src/main/java/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder.java [hunk 2]
```diff
@@ -183,6 +198,21 @@ public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath filePath, @Nonnu
         }
     }
 
+    private void processTests(Run<?, ?> build, FilePath filePath, Launcher launcher, TaskListener listener, TestsToRunConverterResult convertResult) throws IOException, InterruptedException {
```
Comment (nissimshitrit) [inline src/main/java/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder.java:201]: please find meaning name, this method is prepare the input for codeless execution

File: src/main/java/com/microfocus/application/automation/tools/run/RunFromCodelessBuilder.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                build.setResult(Result.FAILURE);
+                taskListener.error("Failed running {} with exception {}", CODELESS_BATCH_FILE, e);
+            } catch (InterruptedException e) {
+                build.setResult(Result.ABORTED);
+                taskListener.error("Failed running {} - build aborted {}", CODELESS_BATCH_FILE, e);
+            }
+        }
+
+        build.setResult(currentResult);
+    }
```
Comment (nissimshitrit) [inline src/main/java/com/microfocus/application/automation/tools/run/RunFromCodelessBuilder.java:88]: please add logs here where its exists for some validations, also if you find other places to add logs it is welcome

**Review by nissimshitrit — CHANGES_REQUESTED** [review]

File: src/main/java/com/microfocus/application/automation/tools/octane/executor/TestExecutionJobCreatorService.java (hunk 1 above)
Comment (itayka) [reply src/main/java/com/microfocus/application/automation/tools/octane/executor/TestExecutionJobCreatorService.java:365]: done

File: src/main/java/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder.java (hunk 2 above)
Comment (itayka) [reply src/main/java/com/microfocus/application/automation/tools/octane/testrunner/TestsToRunConverterBuilder.java:201]: done

File: src/main/java/com/microfocus/application/automation/tools/run/RunFromCodelessBuilder.java (hunk 3 above)
Comment (itayka) [reply src/main/java/com/microfocus/application/automation/tools/run/RunFromCodelessBuilder.java:88]: done

**Review by nissimshitrit — APPROVED** [review]

**Pull request merged**